pkg/upgrade/util.go

```diff
@@ -40,3 +43,17 @@ func ChangeNodeUpgradeState(
 
 	return err
 }
+
+func CreateK8sInterface() (kubernetes.Interface, error) {
+	config, err := rest.InClusterConfig()
```

Comment:
you can use controller runtime ctrl.GetConfigOrDie()
which allows to override kubeconfig via flag or env variable